app/batid/utils/geo.py

```diff
@@ -38,3 +38,27 @@ def fix_nested_shells(geom: GEOSGeometry) -> GEOSGeometry:
         geom = Polygon(big_poly_ring, *small_polys_rings)
 
     return geom
+
+
+def merge_contiguous_shapes(shapes: list):
```

Comment:
j'ai voulu rajouter une protection, au cas ou on essaye de fusionner des bâtiments qui ne se touchent pas.

Reply:
Nous avons un peu plys de 300.000 bâtiments dont la shape est un simple point (`select count(*) from batid_building where ST_GeometryType(shape) = 'ST_Point';`). 

C'est moins de 1% du parc mais je me demande si on ne devrait pas être en mesure de gérer les fusions de ces bâtiments. Je suis un peu partagé sur ce point.

D'un côté, on n'a pas envie de tordre nos méthodes pour des cas exceptionnels, d'un autre on a pas envie non plus de se trainer trop longtemps ces bâtiments point et donc il faut pouvoir les faire rentrer dans nos outils. 

A discuter

Reply:
Je trouve ça bien de ne pas autoriser la fusion "automatique" de bâtiments dont les shapes ne se touchent pas, sinon dans le cas d'un bâtiment point, on va avoir l'union d'un rectangle avec un point à côté comme shape et c'est pas terrible. On peut tout à fait envisager par la suite de rajouter à la fonction `Building.merge()` un paramêtre optionnel de shape, ou bien de dire qu'on doit d'avord updater le batiment point en batiment shape, avant de le fusionner avec un voisin.
On doit pouvoir s'en sortir pas trop mal quand le cas va se présenter.

Reply:
Je suis d'accord qu'un bâtiment fusionné qui deviendrait un polygone + un point n'a pas de sens.
Si on ne prévoit pas la possibilité de fusionner des bâtiments point pour tout de suite, peut-on alors vérifier que tous les bâtiments de liste à fusionner sont des polygones et multipolygones ou lever une exception du style "You can't merge a building which shape is a Point".
Cela permettra à tout le monde d'avoir une compréhension plus fine du pourquoi on ne peut pas fusionner dans ce cas.

Reply:
C'est vrai qu'il est possible que par coup de vraiment pas de chance, on merge un batiment point avec un batiment shape et que le batiment point soit dans le batiment shape. Je corrige avec https://github.com/fab-geocommuns/RNB-coeur/pull/446/commits/8b768b965fc871559b7768f358756d4e012653dd